nixos/modules/services/networking/zerobin.nix

```diff
@@ -0,0 +1,102 @@
+{ config, pkgs, lib, nodes, ... }:
+with lib;
+let
+  cfg = config.services.zerobin;
+
+  zerobin_config = pkgs.writeText "zerobin-config.py" ''
+  PASTE_FILES_ROOT = "${cfg.dataDir}"
+  ${cfg.extraConfig}
+  '';
+
+in
+  {
+    options = {
+      services.zerobin = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+          Enable the 0bin encrypted pastebin
+          '';
+        };
+
+        dataDir = mkOption {
```

Comment:
Perhaps it'd make sense to create/check permissions for this dir in the service's `preStart` hook?